src/sgame/botlib/bot_api.h

```diff
@@ -47,4 +47,12 @@ void         BotDisableArea( const vec3_t origin, const vec3_t mins, const vec3_
 void         BotEnableArea( const vec3_t origin, const vec3_t mins, const vec3_t maxs );
 bool     BotNavTrace( int botClientNum, botTrace_t *trace, const vec3_t start, const vec3_t end );
 
+void BotFindRandomPoint( int botClientNum, vec3_t point );
+bool BotFindRandomPointInRadius( int botClientNum, const vec3_t origin, vec3_t point, float radius );
+
+// be careful: those two function give incorrect behavior if an area is disabled more than once: a node disabled twice will be enabled back on the first try
+// TODO: decide if we want to use G_BotAddObstacle instead
+void G_BotDisableArea( const vec3_t origin, const vec3_t mins, const vec3_t maxs );
+void G_BotEnableArea( const vec3_t origin, const vec3_t mins, const vec3_t maxs );
```

Comment:
I'm kinda pondering if we should remove those, but to be fair, I'm still thinking those are the function which provide the best effect on bots for my tutorial, so I'm keeping those around for a bit longer while I can settle on a proper solution.

Come to think of it, wouldn't `G_BotDisableArea` be really appropriate for a bot clip (disable navmesh) thingy? Though the best solution would probably still be to avoid creating the navmesh there in the first place.

Reply:
> so I'm keeping those around for a bit longer

Yes, that's also what I prefer, or rather, I'd prefer to only keep the non-proxified function, but that's likely something for later.

>  wouldn't G_BotDisableArea be really appropriate for a bot clip

Yes and no. Keep in mind that bot clip still have to be defined. If it is defined as "bots can't go there at all", as the name suggests, then it does not work, since those only prevent bots to *choose* navigating there when not fighting. In a fight, they will ignore that. Same if they're stuck or decide to pounce.

Reply:
Yeah, I probably should call it "nonavmesh" or something like that instead of botclip. I like it when bot can decide to keep attacking even if navmesh don't allow it.